src/core/layout/qgslayoutitemundocommand.cpp

```diff
@@ -55,10 +55,12 @@ void QgsLayoutItemUndoCommand::saveState( QDomDocument &stateDoc ) const
   QDomElement documentElement = stateDoc.createElement( QStringLiteral( "ItemState" ) );
 
   QgsLayoutItem *item = mLayout->itemByUuid( mItemUuid );
-  Q_ASSERT_X( item, "QgsLayoutItemUndoCommand::saveState", "could not retrieve item for saving state" );
-
-  item->writeXml( documentElement, stateDoc, QgsReadWriteContext() );
-  stateDoc.appendChild( documentElement );
+  if ( item )
+  {
+    QgsDebugMsg( QStringLiteral( "QgsLayoutItemUndoCommand::saveState: could not retrieve item %1 for saving state" ).arg( mItemUuid ) );
```

Comment:
shouldn't this debug message be in an else clause, that is when item == nullptr ?

Reply:
O course, thank you for noticing!